recipes/nasm/all/conanfile.py

```diff
@@ -11,85 +18,100 @@ class NASMConan(ConanFile):
-        if str(self.version).startswith("2.13"):
-            tools.replace_in_file("Makefile", "$(CC) $(LDFLAGS) -o", "$(CC) $(ALL_CFLAGS) $(LDFLAGS) -o")
+        # TODO I assume this is required
+        # inject requires env vars in build scope
+        # it's required in case of native build when there is AutotoolsDeps & at least one dependency which might be shared, because configure tries to run a test executable
+        if not cross_building(self):
+            env = VirtualRunEnv(self)
+            env.generate(scope="build")
 
-        return self._autotools
 
     def build(self):
-        for patch in self.conan_data.get("patches", {}).get(self.version, []):
-            tools.patch(**patch)
-        if self.settings.compiler == "Visual Studio":
-            self._build_vs()
+        apply_conandata_patches(self)
+
+        if is_msvc(self):
+            with chdir(self, self.source_folder):
+                self.run("nmake /f {}".format(os.path.join("Mkfiles", "msvc.mak")))
+                shutil.copy("nasm.exe", "nasmw.exe")
+                shutil.copy("ndisasm.exe", "ndisasmw.exe")
         else:
-            autotools = self._configure_autotools()
+            autotools = Autotools(self)
+            autotools.configure()
+
+            # GCC9 - "pure" attribute on function returning "void"
+            replace_in_file(self, "Makefile", "-Werror=attributes", "")
+
+            # Need "-arch" flag for the linker when cross-compiling.
+            # FIXME: Revisit after https://github.com/conan-io/conan/issues/9069, using new Autotools integration
+            # TODO it is time to revisit, not sure what to do here though...
+            if str(self.version).startswith("2.13"):
+                replace_in_file(self, "Makefile", "$(CC) $(LDFLAGS) -o", "$(CC) $(ALL_CFLAGS) $(LDFLAGS) -o")
+                replace_in_file(self, "Makefile", "$(INSTALLROOT)", "$(DESTDIR)")
             autotools.make()
 
+
     def package(self):
-        self.copy(pattern="LICENSE", src=self._source_subfolder, dst="licenses")
-        if self.settings.compiler == "Visual Studio":
-            self.copy(pattern="*.exe", src=self._source_subfolder, dst="bin", keep_path=False)
+        copy(self, pattern="LICENSE", dst=os.path.join(self.package_folder, "licenses"), src=self.source_folder)
+        if is_msvc(self):
+            copy(self, pattern="*.exe", src=self.source_folder, dst=os.path.join(self.package_folder, "bin"), keep_path=False)
         else:
-            autotools = self._configure_autotools()
+            autotools = Autotools(self)
             autotools.install()
-            tools.rmdir(os.path.join(self.package_folder, "share"))
+            rmdir(self, os.path.join(self.package_folder, "share"))
 
     def package_info(self):
         bin_path = os.path.join(self.package_folder, "bin")
         self.output.info("Appending PATH environment variable: {}".format(bin_path))
         self.env_info.PATH.append(bin_path)
```

Comment:
This could be under an "if conan_version < 2" check so we can test the v2 pipeline .... Strawberry perl was updated and that's the only requirement.

It's an extra but i'd love to take the time to make sure it's passing before OpenSSL 🙊

Reply:
I've added the test, copied from ninja's recipe, is this what you meant?

Reply:
No. It means `self.env_info` is not Conan v2 compatible. So, to run Conan v2 with this recipe, you need to update it to:

```python
from conan import conan_version

...

def package_info(self):
    if Version(conan_version).major < 2:
        bin_path = os.path.join(self.package_folder, "bin")
        self.output.info(f"Appending PATH environment variable: {bin_path}")
        self.env_info.PATH.append(bin_path)
```

Reply:
Ok, I made it as you wrote

Reply:
>  I assumed the libdirs and includedirs=[] remain outside the if block

Yes, for both Conan v1 and v2 they should persist.


About that `if Version(conan_version).major < 2:` will no longer be needed as on Conan 2.0 beta 5, `self.env_info` will be mocked.

Reply:
ok